src/field.ts

```diff
@@ -87,11 +86,11 @@ class Field {
                     ? fieldObj.schema.$uid
                     : fieldObj.uid;
             const path = schemaPath.split(".");
-            let value = event.data;
-
+            let value: GenericObjectType = new Map(Object.entries(event.data));
```

Comment:
Why is this needed? Instead of using GenericObjectType either declare it as indexable or something with data & indexable will solve the problem. This solution doesn't make sense.